### ISSUE TITLE
Add support for Julia v0.7 and v1.0

### ISSUE THREAD

**Issue by YongHee-Kim** [felipenoris/XLSX.jl#46]
I'm on 'XLSX#master' but precomfile fails on MAC (`OS: macOS (x86_64-apple-darwin14.5.0`)

``` julia
(v0.7) pkg> precompile
Precompiling project...
Precompiling XLSX
[ Info: Precompiling XLSX [fdbf4ff8-1666-58a4-91e7-1b58723a45e0]
┌ Warning: `method_exists(f, t)` is deprecated, use `hasmethod(f, t)` instead.
│   caller = top-level scope at none:0
└ @ Core none:0
WARNING: importing deprecated binding Base.Dates into XLSX.
WARNING: Base.Dates is deprecated, run `using Dates` instead
  likely near /Users/devsisters/.julia/packages/XLSX/kKqNW/src/structs.jl:45
WARNING: Base.Dates is deprecated, run `using Dates` instead
  likely near /Users/devsisters/.julia/packages/XLSX/kKqNW/src/structs.jl:45
WARNING: Base.Dates is deprecated, run `using Dates` instead
  likely near /Users/devsisters/.julia/packages/XLSX/kKqNW/src/structs.jl:45
WARNING: importing deprecated binding Base.Nullable into XLSX.
ERROR: LoadError: LoadError: TypeError: in Type{...} expression, expected UnionAll, got typeof(Base.Nullable)
Stacktrace:
```

**Comment by YongHee-Kim** [comment]
I saw #v0.3.0 release. And did some dirty fix to working it on 0.7.0 [on my fork](https://github.com/YongHee-Kim/XLSX.jl) but couldn't make it run on 1.0. I know you are working on this, but if you want any help I will make PR 

list of my fixes are...
1. replace `Nullable` with `Union{Missing, T}`
2. fix runtest.jl to run on 0.7 (somehow `pkg>test XLSX` doesn't work, but it works on REPL)
3. replace `Base.start, Base.next, Base.done` with `start, next, done`

**Comment by felipenoris** [comment]
Yes. Please check out branch v0.7 . I think most of the work is done. If not, feel free to contribute with more fixes.
I guess the new iterator protocol is deprecated in v0.7, but not allowed in v1.0. I'll be working on that in a few days.
